### DISCUSSION TITLE
Appdata is not present. after migration

### DISCUSSION THREAD

**Discussion by LukeVader-IV** [nextcloud/all-in-one#3664] [Questions] [answered]
after following [migration.md - Migrate the files and the database](https://github.com/nextcloud/all-in-one/blob/main/migration.md#migrate-the-files-and-the-database) and starting up i get the followinng error:
```
Installing imagemagick via apk...
Enabling Imagick...
WARNING: opening from cache https://dl-cdn.alpinelinux.org/alpine/v3.18/main: No such file or directory
WARNING: opening from cache https://dl-cdn.alpinelinux.org/alpine/v3.18/community: No such file or directory
Configuring Redis as session handler...
Setting php max children...
Appdata is not present. Did you maybe change the datadir after the initial Nextcloud installation? This is not supported!
See https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir
If you adjusted the datadir to be located on an external drive, make sure that the drive is still mounted!
In the datadir was found:
total 1936
drwxr-x--- 1 www-data root     526 Nov  2 14:24 .
drwxr-xr-x 1 root     root      12 Nov  2 14:24 ..
-rwxr-x--- 1 www-data root     542 Nov  2 13:28 .htaccess
-rwxr-x--- 1 www-data root       0 Nov  2 13:28 .ocdata
-rwxr-x--- 1 www-data root     101 Oct 26 17:21 .user.ini
drwxr-x--- 1 www-data root     858 Oct 30 19:31 3rdparty
-rwxr-x--- 1 www-data root   23796 Oct 26 17:21 AUTHORS
-rwxr-x--- 1 www-data root   34520 Oct 26 17:21 COPYING
drwxr-x--- 1 www-data root    1694 Oct 30 19:31 apps
-rwxr-x--- 1 www-data root    1283 Oct 26 17:21 composer.json
-rwxr-x--- 1 www-data root    3140 Oct 26 17:21 composer.lock
drwxr-x--- 1 www-data root      72 Oct 30 19:31 config
-rwxr-x--- 1 www-data root    4095 Oct 26 17:21 console.php
drwxr-x--- 1 www-data root     480 Oct 30 19:31 core
-rwxr-x--- 1 www-data root    6317 Oct 26 17:21 cron.php
drwxr-x--- 1 www-data root     558 Apr  1  2023 data
drwxr-x--- 1 www-data root   11356 Oct 30 19:31 dist
-rwxr-x--- 1 www-data root       0 Nov  2 13:28 index.html
-rwxr-x--- 1 www-data root    3456 Oct 26 17:21 index.php
drwxr-x--- 1 www-data root     126 Oct 30 19:31 lib
-rwxr-x--- 1 www-data root     283 Oct 26 17:21 occ
drwxr-x--- 1 www-data root      50 Oct 30 19:31 ocs
drwxr-x--- 1 www-data root      18 Oct 30 19:31 ocs-provider
-rwxr-x--- 1 www-data root 1849208 Oct 26 17:21 package-lock.json
-rwxr-x--- 1 www-data root    6342 Oct 26 17:21 package.json
-rwxr-x--- 1 www-data root    3187 Oct 26 17:21 public.php
-rwxr-x--- 1 www-data root    5597 Oct 26 17:21 remote.php
drwxr-x--- 1 www-data root     158 Oct 30 19:31 resources
-rwxr-x--- 1 www-data root      26 Oct 26 17:21 robots.txt
-rwxr-x--- 1 www-data root    2452 Oct 26 17:21 status.php
drwxr-x--- 1 www-data root      26 Oct 30 19:31 themes
drwxr-x--- 1 www-data root      42 Oct 30 19:31 updater
-rwxr-x--- 1 www-data root     403 Oct 26 17:25 version.php
              now              
-------------------------------
 2023-11-02 14:24:33.160061+01
(1 row)

```
from my understandng this should be correct? i copied all the data from my old server to my new one, am i missing something?
every single step of the migration guide was carefully followed.
the only thing i can think of is that it doesn't expect /mnt/ncdata/{the data} but rather something like /mnt/ncdata/Appdata/{the data}. but if so, that isn't very clear

**Comment by szaimen** [comment, marked as answer]
> drwxr-x--- 1 www-data root     558 Apr  1  2023 data

Hi indeed, should the contents of this folder which is called the datadir, be moved to the datadir folder. Not the stuff that is in the parent folder. Improvements to the documentation are very welcome!

**Reply by LukeVader-IV** [reply]
ah right, i was confused as i thought the directory which on my old server was /var/www/nextcloud/ needed to be copied, instead it was /var/www/nextcloud/data/.
In retrospect it makes sense as data in it's parent directory really shouldn't need to be copied, only userdata and appdata are critical to be copied over, thank you.

i checked the documentation and it is specified that it should be the data subdir, and not the entirety of the subdirectory
however it's notation is unclear, `/path/to/nextcloud/data/.` the first part would imply to read it as a sentance, like "path to nextcloud data" which i understood to be the parent directory. i think `/path/to/` is not the right thing to use in this scenario.
However i'm not sure exactly how it should be written, but an idea might be to write `.../.../nextcloud/data/.` to show that the data part is to be taken literally. Or maybe simply write `/nextcloud/data/.` but i'm not a fan of that as that could also potentially cause confusion.
